deezone/HydroBytes-BaseStation
# Issue-61 Row level authorization

resolves #61                   

### Description

This PR adds row level authorization. Only when a row in a database table is owned by the account or the account has the admin role can the row be altered.

### How to Test

- [x] confirm `migrate` and `seed` work and include changes to `station` table
```
> go run ./cmd/admin migrate
Migrations complete

> go run ./cmd/admin seed
Seed data complete
```

- confirm api endpoints control access based on the account (`Admin` vs `Station 000x` vs none) JWT value used to access the endpoint:
  - [x] no token - `GET /v1/station-types`
  - [x] requires `Admin` role - POST /v1/station-type:
  - [x] is not owner - `PUT /v1/station/{id}`

**`403` Forbidden**
```
{
    "error": "expected authorization header format: Bearer <token>"
}
```

- [x] confirm all tests are passing:
```
> go test ./internal/station_type
ok  	github.com/deezone/HydroBytes-BaseStation/internal/station_type	26.168s

> go test ./cmd/api/tests/account_tests
ok  	github.com/deezone/HydroBytes-BaseStation/cmd/api/tests/account_tests	3.267s

> go test ./cmd/api/tests/station_tests
ok  	github.com/deezone/HydroBytes-BaseStation/cmd/api/tests/station_tests	3.222s

go test ./cmd/api/tests/station_type_tests
ok  	github.com/deezone/HydroBytes-BaseStation/cmd/api/tests/station_type_tests	3.123s
```
